relays/messages/src/message_race_delivery.rs

```diff
@@ -387,13 +387,11 @@ where
 		race_state: &mut RaceState<SourceHeaderIdOf<P>, TargetHeaderIdOf<P>, P::MessagesProof>,
 	) {
 		// best target nonces must always be ge than finalized target nonces
-		let mut target_nonces = self.target_nonces.take().unwrap_or_else(|| nonces.clone());
```

Comment:
This change here and similar change in the other file comes from the #719, but IIUC this (`cmp::max`) should only happen in the **finalized** version of those methods.